### ISSUE TITLE
Invoke hangs in input when using in_stream to pass data to crontab

### ISSUE THREAD

**Issue by gertjanol** [pyinvoke/invoke#602]
I'm trying to manipulate a users crontab with an invoke task. Direct access to a crontab is not allowed, so you have to use the `crontab`-tool. The process is simple: use `crontab -l` to read the contents. manipulate it and write is back using (basically) `echo "new-contents" | crontab`.
When using the `in_stream`-parameter to `ctx.run` with the new contents (in a `StringIO` object), the invoke task hangs. Using strace on the process I can see that it's waiting on input.

A simple implementation using `subprocess.Popen` works perfectly.

It's very easy to reproduce (**this will overwrite your local crontab, make sure it doesn't contain anything important!)**:
```
$ mkdir invoke-test
$ cd invoke-test
$ mkvirtualenv -ppython2.7 invoke-test
$ pip install invoke==1.2.0

$ cat <<EOM>tasks.py
from StringIO import StringIO
from subprocess import Popen, PIPE

from invoke import task

crontab = '* * * * * echo shibby\n'


@task
def hangs(c):
    c.run('crontab', in_stream=StringIO(crontab))


@task
def works(c):
    p = Popen(['crontab'], stdin=PIPE, stdout=PIPE, stderr=PIPE)
    _, _ = p.communicate(crontab)
    p.poll()
EOM
```

```
$ inv -l
Available tasks:

  hangs
  works

$ inv works
$ crontab -l
* * * * * echo shibby
$ crontab -r
$ inv hangs
<nothing>
```

Python 2 (.7) or Python 3 (.6) makes no difference. Also the fix-everything-pty=true trick didn't help.

Maybe I don't understand how to use the `in_stream` parameter or this may not be the intended use, but I'd love some feedback!

**Comment by mathieulongtin** [comment]
It appears that run() doesn't close stdin once the in_stream is at EOF. You can test this trivially:
```
run("cat -n", in_stream=io.BytesIO(b"some\nlines\n"))
```
This will block. 

The other weird thing is that it appears that it feeds stdin one character at a time.

**Comment by bitprophet** [comment]
This feels like a duplicate of #552, rolling into that.
